src/ixdat/measurements.py

```diff
@@ -264,26 +398,291 @@ def get_t_and_v(self, item, tspan=None):
+                if False not in mask:
+                    new_series_list.append(series)
+                elif series.identity == t_identity:
+                    new_series_list.append(new_tseries)
+                else:
+                    new_series = series.__class__(
+                        name=series.name,
+                        unit_name=series.unit_name,
+                        data=series.data[mask],
+                        tseries=new_tseries,
+                    )
+                    new_series_list.append(new_series)
+        obj_as_dict["series_list"] = new_series_list
+
+        # then cut the component measurements.
+        new_component_measurements = []
+        for m in self._component_measurements:
+            # FIXME: This is perhaps overkill, to make new cut component measurements,
+            #    as it duplicates data (a big no)... especially bad because
+            #    new_measurement.save() saves them.
+            #    The step is here in order for file_number to get built correctly.
+            dt = m.tstamp - self.tstamp
+            tspan_m = [tspan[0] - dt, tspan[1] - dt]
+            if m.tspan[-1] < tspan_m[0] or tspan_m[-1] < m.tspan[0]:
+                continue
+            new_component_measurements.append(m.cut(tspan_m))
+        obj_as_dict["component_measurements"] = new_component_measurements
+
+        new_measurement = self.__class__.from_dict(obj_as_dict)
+        return new_measurement
+
+    def select_value(self, *args, **kwargs):
+        """Return a new Measurement with the time(s) meeting criteria.
+
+        Can only take one arg or kwarg!
+        The `series_name` is `self.sel_str` if given an arg, kw if given a kwarg.
+        Either way the argument is the `value` to be selected for.
+
+        The method finds all time intervals for which `self[series_name] == value`
+        It then cuts the measurement according to each time interval and adds these
+        segments together. TODO: This can be done better, i.e. without chopping series.
+
+        TODO: greater-than and less-than kwargs.
+            Ideally you should be able to say e.g., `select(cycle=1, 0.5<potential<1)`
+        """
+        if len(args) >= 1:
+            if not self.sel_str:
+                raise BuildError(
+                    f"{self} does not have a default selection string "
+                    f"(Measurement.sel_str), and so selection only works with kwargs."
+                )
+            kwargs[self.sel_str] = args
+        if len(kwargs) > 1:
+            raise BuildError(
+                f"select_value got kwargs={kwargs} but can only be used for one value "
+                f"at a time. Use select_values for more."
+            )
+        new_measurement = self
+        ((series_name, value),) = kwargs.items()
+
+        t, v = self.grab(series_name)
+        mask = v == value  # linter doesn't realize this is a np array
+        mask_prev = np.append(False, mask[:-1])
```

Comment:
This is I think it's fair to say absolutely not trivial. I'm able to write this stuff fluently (so long as it's one dimension, otherwise I can never remember the orders of the dimensions and reshaping etc syntax) only due to practice in Matlab (where this stuff reads a bit more intuitively) AND (often) debugging. 

I will add explanatory comments! If an ixdat user and documentation reader learns tricks with vectorized numerical data analysis from the comments it will be well worth the time 😄

Reply:
done.